ModernWallet/Classes/ViewController/Buy/BuyOptimizedViewController.swift

```diff
@@ -280,47 +305,47 @@ fileprivate extension ObservableType where Self.E == Void {
+            errors.asObservable().bind(to: viewController.rx.error),
+            success.drive(onNext: {[weak viewController] _ in
+                viewController?.buyOptimizedViewModel.buyAmount.value = BuyOptimizedViewModel.Amount(autoUpdated: true, value: "", showErrorMsg: false)
+                viewController?.buyOptimizedViewModel.payWithAmount.value = BuyOptimizedViewModel.Amount(autoUpdated: true, value: "", showErrorMsg: false)
             }),
-            success.drive(onNext: {[weak vc] _ in
+            success.drive(onNext: {[weak viewController] _ in
                 FinalizePendingRequestsTrigger.instance.finalizeNow()
-                vc?.view.makeToast("Your exchange has been successfuly processed.It will appear in your transaction history soon.")
+                viewController?.view.makeToast(Localize("buy.newDesign.sucess"))
             })
         ]
     }
 }
 
 fileprivate extension BuyOptimizedViewModel {
     
-    func bind(toViewController vc: BuyOptimizedViewController) -> [Disposable] {
+    func bind(toViewController viewController: BuyOptimizedViewController) -> [Disposable] {
         return [
             isValidPayWithAmount
                 .filterError()
                 .map{ $0["Message"] as? String }
                 .filterNil()
-                .subscribe(onNext: { [weak vc] message in
-                    vc?.view.makeToast(message, duration: 2.0, position: CSToastPositionTop)
+                .subscribe(onNext: { [weak viewController] message in
+                    viewController?.view.makeToast(message, duration: 2.0, position: CSToastPositionTop)
                 }),
-            isValidPayWithAmount.map{ $0.isSuccess }.bind(to: vc.submitButton.rx.isEanbledWithBorderColor),
-            spreadPercent.drive(vc.spreadPercent.rx.text),
-            spreadAmount.drive(vc.spreadAmount.rx.text),
-            bid.asDriver().filterNil().map{ $0 ? "SELL" : "PAY WITH" }.drive(vc.walletListView.label.rx.text),
-            bid.asDriver().filterNil().map{ $0 ? "RECEIVE" : "BUY" }.drive(vc.assetListView.label.rx.text),
-            bid.asDriver().filterNil().map{ $0 ? "SELL" : "BUY" }.drive(onNext: {[weak vc] in
-                vc?.submitButton.setTitle($0, for: .normal)
+            isValidPayWithAmount.map{ $0.isSuccess }.bind(to: viewController.submitButton.rx.isEanbledWithBorderColor),
+            spreadPercent.drive(viewController.spreadPercent.rx.text),
+            spreadAmount.drive(viewController.spreadAmount.rx.text),
+            bid.asDriver().filterNil().map{ $0 ? "SELL" : "PAY WITH" }.drive(viewController.walletListView.label.rx.text),
+            bid.asDriver().filterNil().map{ $0 ? "RECEIVE" : "BUY" }.drive(viewController.assetListView.label.rx.text),
+            bid.asDriver().filterNil().map{ $0 ? "SELL" : "BUY" }.drive(onNext: {[weak viewController] in
```

Comment:
You can use the following drive: `.drive(viewController?.submitButton.rx.title(for: .normal)` instead of `onNext:` closure